core/internal/cltest/factories.go

```diff
@@ -540,8 +540,7 @@ func MustInsertKeeperJob(t *testing.T, db *sqlx.DB, korm keeper.ORM, from ethkey
 	require.NoError(t, err)
 
 	var pipelineSpec pipeline.Spec
-	dds := keeper.ExpectedObservationSource
-	err = korm.Q().Get(&pipelineSpec, `INSERT INTO pipeline_specs (dot_dag_source,created_at) VALUES ($1,NOW()) RETURNING *`, dds)
+	err = korm.Q().Get(&pipelineSpec, `INSERT INTO pipeline_specs (dot_dag_source,created_at) VALUES ('',NOW()) RETURNING *`)
```

Comment:
maybe this can be deleted completely?

Reply:
From my understanding, the job still has to include a pipeline spec, even if it is blank (which is our case now), as we are injecting the observation source in the upkeep executor.